spec/fitbank_api/pix/dynamic_qr_code_spec.rb

```diff
@@ -0,0 +1,48 @@
+# typed: false
+# frozen_string_literal: true
+
+require 'date'
+require 'bigdecimal'
+require 'securerandom'
+
+RSpec.describe FitBankApi::Pix::DynamicQrCode do
+  describe 'Dynamic QR code' do
+
+    before(:all) do
+      VCR.use_cassette('pix/qr_code/dynamic/generate') do
+        latam_bank_info = build(:bank_info)
+        credentials = build(:credentials)
+
+        @qr_code_manager = described_class.new(
+          base_url: ENV.fetch('FITBANK_BASE_URL'),
+          receiver_bank_info: latam_bank_info,
+          receiver_pix_key: credentials.cnpj,
+          credentials: credentials,
+          receiver_zip_code: ENV.fetch('LATAM_ZIP_CODE')
+        )
+
+        @generate_response = @qr_code_manager.generate(
+          value: BigDecimal('20'),
+          expiartion_date: Date.today + 30,
+          id: SecureRandom.uuid
+        )
+      end
+    end
+
+    it 'can generate dynamic QR code' do
+      expect(@generate_response[:Success]).to eq('true')
+      expect(@generate_response).to have_key(:DocumentNumber)
+    end
```

Comment:
Why are we using a VCR cassette in the `before` block?
The test structure looks odd.

Reply:
The idea of the test is to generate a qr code and the query it. In the before block we generate it. In the the test we check if the response has the correct form and in the other test we check if the code can be found. The thing is that we need to keep state and the state is the DocumentNumber returned by generate.